### PR TITLE
genoconvert

### PR DESCRIPTION
**Pull request by nevrome**
As suggested and discussed already a loooooong time ago in #12.

This is a minimal implementation that just replaces the old genotype data with a converted version.

### CODE REVIEW

**Comment by stschiff** [comment]
OK great. I would only suggest to add one more option to control the removing-files behaviour. In fact, I would prefer to keep the old format files by default and only remove them if the user specifies that with `--remove-old` or so.
A second comment concerns the checksums. We currently don't create those, neither for `forge` nor for this new subcommand. Shouldn't we do that by default? I mean, when if not upon genotype-data creation should we compute checksums?

**Comment by nevrome** [comment]
Thank you for checking!

Regarding keeping the old files: As I already stated [here](https://github.com/poseidon-framework/poseidon-hs/issues/12#issuecomment-786200604) I don't know then what should happen in case of a back-conversion. Should we overwrite then?

And regarding the checksums I made my point [here](https://github.com/poseidon-framework/poseidon-hs/issues/35#issuecomment-778679298): I think they are an advanced feature and only really relevant for our repository. We also loosened checksum tests to speed up trident, so many users might not get any value out of them. BUt maybe I'm missing an important usecase :man_shrugging: ?

**Comment by stschiff** [comment]
Right, I think it's still the right decision not to check checksums by default, but they're definitely a plus. They give extra security that a package contains specific data. But I have to admit I also don't recognise an immediate benefit for the user... so I'm not strong about that.

With respect to switching to old data... I think we should simply print out an error if we want to write files that are already there. Frankly, if you already have genotype files in the other format, then you don't need genoconvert, you can just manually change the yaml file.

Let me propose that change as a commit and see how you like it.

**Comment by stschiff** [comment]
OK, still need to put in a check for the output files.

File: src-executables/Main-trident.hs [hunk 1]
```diff
@@ -163,6 +163,10 @@ forgeOptParser = ForgeOptions <$> parseBasePaths
 genoconvertOptParser :: OP.Parser GenoconvertOptions
 genoconvertOptParser = GenoconvertOptions <$> parseBasePaths
                                           <*> parseOutGenotypeFormat
+                                          <*> parseRemoveOld
+
+parseRemoveOld :: OP.Parser Bool
+parseRemoveOld = OP.switch (OP.long "remove-old" <> OP.help "Remove the old genotype files when creating the new ones")
```
Comment (nevrome) [inline src-executables/Main-trident.hs:169]: Afaik all our option names are in camelCase at the moment - so maybe `removeOld` would fit in better than `remove-old`

Comment (stschiff) [reply src-executables/Main-trident.hs:169]: Ah ok. Fine. I agree

**Review by stschiff — APPROVED** [review]

**Pull request merged**